api/v1alpha1/hostedcluster_types.go

```diff
@@ -82,6 +82,11 @@ const (
 	// This is a temporary workaround necessary for compliance reasons on the IBM Cloud side:
 	// no images can be pulled from registries outside of IBM Cloud's official regional registries
 	MachineApproverImage = "hypershift.openshift.io/machine-approver-image"
+
+	// PauseReconciliationAnnotation is an annotation that can be used to pause reconciliation on resources. Either
+	// a date can be provided in RFC3339 format or a boolean. If a date is provided: reconciliation is paused until that date.
```

Comment:
what are scenarios where we need date and changing a boolean is not enough?

Reply:
The SRE team brought the request up for date in this thread:
https://coreos.slack.com/archives/C01C8502FMM/p1645645426109709?thread_ts=1645643483.593509&cid=C01C8502FMM

The idea is with pause being a "manual thing" as far as setting the annotation date provides a way where if someone forgets operations aren't impacted for forever:

how could I manually change config if the controller is paused? is that something that could be directly managed?

[Naveen Malik](https://app.slack.com/team/UBCBRKRK4)  [6 days ago](https://coreos.slack.com/archives/C01C8502FMM/p1645645478889529?thread_ts=1645643483.593509&cid=C01C8502FMM)
suggestion on pause, maybe a "pause until date" because when we pause it's a manual thing and humans forget. forgetting a cluster is paused is a Bad Thing